authors/apps/friends/views.py

```diff
@@ -0,0 +1,120 @@
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from authors.apps.authentication.serializers import UserSerializer
+from .models import Friend
+from .serializers import CustomUserSerializer
+
+
+class FollowUnfollowApiView(generics.RetrieveUpdateDestroyAPIView):
+    """view for following a user"""
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, username, format=None):
+        """
+        Checks if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, returned error message
+        Else create the user relationship
+        Finally return the serialized object of the followed user
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot follow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+        Friend.objects.get_or_create(user_from=follower, user_to=followed)
```

Comment:
What happens if i try to follow a user that i have followed before?

Reply:
For now, no error is returned but the followers count of that user remains the same. What would be the best way to handle that?

Reply:
I have now worked on this. A user will get a message saying they already follow the user

Reply:
Since you are using get_or_create, if it gets the value and `created` returns False, then return a Response that user has already being followed. See https://docs.djangoproject.com/en/2.1/ref/models/querysets/#get-or-create
In that case, you return early and do not have to execute lines 37 - 41